Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,943 @@
+// Copyright (c) 2024 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Roberto Dyke, Sven Oesau
+//
+#ifndef CGAL_POLYGON_MESH_PROCESSING_REGISTER_MESH_H
+#define CGAL_POLYGON_MESH_PROCESSING_REGISTER_MESH_H
+
+#include <CGAL/license/Polygon_mesh_processing/registration.h>
+
+#include <CGAL/config.h>
+
+#ifdef CGAL_EIGEN3_ENABLED
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <Eigen/SVD>
+#include <Eigen/Sparse>
+#endif
+
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/boost/graph/Face_filtered_graph.h>
+
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Surface_mesh_deformation.h>
```

Comment:
is this one really needed? Seems you are only using the closest_rotation_traits, no?

Reply:
The ARAP visitor is defined in that header. However, it only uses a few lines of code and the Cotangent_weight. We could remove the dependency to Surface_mesh_deformation.

Reply:
That was indeed my idea, moving the thing in Solver_interface or Weight package